blocks/library/code-mirror/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import CodeMirror from 'react-codemirror';
+import he from 'he';
+require( 'codemirror/mode/css/css' );
+require( 'codemirror/mode/diff/diff' );
+require( 'codemirror/mode/javascript/javascript' );
+require( 'codemirror/mode/markdown/markdown' );
+require( 'codemirror/mode/pegjs/pegjs' );
+require( 'codemirror/mode/php/php' );
+require( 'codemirror/lib/codemirror.css' );
+
+/**
+ * Internal dependencies
+ */
+import { registerBlockType, query } from '../../api';
+const { html } = query;
+
+const toEditor = s => he.decode( s.split( '<br>' ).join( '\n' ) );
+const fromEditor = s => he.encode( s, { useNamedReferences: true } ).split( '\n' ).join( '<br>' );
```

Comment:
>(note also that `he` has a few cool points, like being able to choose how and when characters get encoded)

It's also a behemoth of a library, at least 24kb minified + gzipped. Might be a point toward dynamically enqueueing block scripts separately based on presence in content.